insights/core/serde.py

```diff
@@ -196,27 +196,34 @@ def dehydrate(self, comp, broker):
                 fs.ensure_path(self.data, mode=0o770)
             self.created = True
 
-        c = comp
-        doc = None
         try:
-            name = dr.get_name(c)
+            name = dr.get_name(comp)
+
             # The `broker.trackbacks` is a dict in which the values are string
             # but not list of strings
-            errors = [broker.tracebacks[e] for e in broker.exceptions.get(c, [])]
-            doc = {
-                "name": name,
-                "exec_time": broker.exec_times.get(c),
-                "errors": errors
-            }
+            errors = [broker.tracebacks[e] for e in broker.exceptions.get(comp, [])]
+
+            # For datasource timeouts the key is the name not the component, so
+            # check for the name in the exception dict.
+            if broker.exceptions.get(name):
+                errors.extend([broker.tracebacks[e] for e in broker.exceptions.get(name, [])])
```

Comment:
Maybe nitpicking, but how about?
```
exs = broker.exceptions
errors = [broker.tracebacks[e] for e in exs.get(comp, exs.get(name, []))]
```

Reply:
Good nitpick, I was trying to figure out a way to better optimize it, and I hadn't thought about putting the second get call in the default of the first. Thanks, I updated it.